src/rules/no-unsafe-values.js

```diff
@@ -25,6 +31,48 @@ export default {
 						loc: node.loc,
 						messageId: "unsafeNumber",
 					});
+				} else {
+					const txt = context.sourceCode.getText(node);
+					if (node.value === 0) {
+						// If the value has been rounded down to 0, but there was some fraction
+						// or non-zero part before an e-, this is a very small number that doesn't
+						// fit inside an f64.
+						for (const r of SMALL_NUMBERS) {
+							const m = txt.match(r);
+							// If the digits are all 0, it's ok.
+							if (m?.[1]?.match(/[1-9]/u)) {
```

Comment:
Just eliminating one-character variables.
```suggestion
						for (const pattern of SMALL_NUMBERS) {
							const match = txt.match(pattern);
							// If the digits are all 0, it's ok.
							if (match?.[1]?.match(/[1-9]/u)) {
```